orchestrator/schema/experiment.py

```diff
@@ -624,69 +624,72 @@ def validate_entity(self, entity: "Entity", strict_optional=False) -> bool:
         """Returns True if Experiment can be applied to entity, false otherwise
 
         This method only checks constitutive properties.
-        - All properties of the Entity that match the experiments required or optional properties must have
-        values in the domain of that property
-        - All required properties of the experiment must have a matching constitutive property
-        - If strict_optional is True all properties of the Entity that are not required properties of the Experiment
-        must match optional properties of the experiment.
+        - The entity has valid values for all required properties of the experiment
+        - The entity has valid values for any optional properties of the experiment it contains
+        - If strict_optional is True all properties of the Entity are properties (required+optional) of the experiment
```

Comment:
I don't think `strict_optional` really conveys the intent behind this flag, I would say this is more like `disallow_extra_properties` - if that is true, there cannot be properties that are not constitutive or optional properties of the experiment.